File: sdks/python/bolt_prediction_sdk/models/report_statistics_schema.py
"""
ReportStatisticsSchema 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class ReportStatisticsSchema(SDKBaseModel):
    """报告统计数据"""

    prediction_count: Optional[int] = Field(description="预测次数", default=0)
    avg_risk_score: Optional[float] = Field(description="平均风险评分", default=0.0)
    min_risk_score: Optional[float] = Field(description="最低风险评分（最高风险）", default=0.0)
    max_risk_score: Optional[float] = Field(description="最高风险评分（最低风险）", default=0.0)
    status_distribution: Optional[Dict[str, int]] = Field(description="状态分布", default=None)
    trend: Optional[str] = Field(description="整体趋势", default='stable')
    max_status: Optional[str] = Field(description="周期内最高状态", default='正常')
    fault_types: Optional[List[str]] = Field(description="出现的故障类型", default=None)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/ReportStatisticsSchema.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 报告统计数据 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class ReportStatisticsSchema {

    @JsonProperty("prediction_count")
    private Integer predictionCount;

    @JsonProperty("avg_risk_score")
    private Double avgRiskScore;

    @JsonProperty("min_risk_score")
    private Double minRiskScore;

    @JsonProperty("max_risk_score")
    private Double maxRiskScore;

    @JsonProperty("status_distribution")
    private Map<String, Integer> statusDistribution;

    @JsonProperty("trend")
    private String trend;

    @JsonProperty("max_status")
    private String maxStatus;

    @JsonProperty("fault_types")
    private List<String> faultTypes;

    public Integer getPredictionCount() {
        return predictionCount;
    }

    public void setPredictionCount(Integer predictionCount) {
        this.predictionCount = predictionCount;
    }

    public Double getAvgRiskScore() {
        return avgRiskScore;
    }

    public void setAvgRiskScore(Double avgRiskScore) {
        this.avgRiskScore = avgRiskScore;
    }

    public Double getMinRiskScore() {
        return minRiskScore;
    }

    public void setMinRiskScore(Double minRiskScore) {
        this.minRiskScore = minRiskScore;
    }

    public Double getMaxRiskScore() {
        return maxRiskScore;
    }

    public void setMaxRiskScore(Double maxRiskScore) {
        this.maxRiskScore = maxRiskScore;
    }

    public Map<String, Integer> getStatusDistribution() {
        return statusDistribution;
    }

    public void setStatusDistribution(Map<String, Integer> statusDistribution) {
        this.statusDistribution = statusDistribution;
    }

    public String getTrend() {
        return trend;
    }

    public void setTrend(String trend) {
        this.trend = trend;
    }

    public String getMaxStatus() {
        return maxStatus;
    }

    public void setMaxStatus(String maxStatus) {
        this.maxStatus = maxStatus;
    }

    public List<String> getFaultTypes() {
        return faultTypes;
    }

    public void setFaultTypes(List<String> faultTypes) {
        this.faultTypes = faultTypes;
    }

}